### PR TITLE
openMP accelerating DeepMD computing in ABACUS

### PR DESCRIPTION
**Pull request by HaCTang**
### Reminder
- [ ] Have you linked an issue with this pull request?
- [ ] Have you added adequate unit tests and/or case tests for your pull request?
- [ ] Have you noticed possible changes of behavior below or in the linked issue?
- [ ] Have you explained the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi? (ignore if not applicable)

### Linked Issue
Fix #...

### Unit Tests and/or Case Tests for my changes
- A unit test is added for each new feature or bug fix.

### What's changed?
- Example: My changes might affect the performance of the application under certain conditions, and I have tested the impact on various scenarios...

### Any changes of core modules? (ignore if not applicable)
- Example: I have added a new virtual function in the esolver base class in order to ...

### CODE REVIEW

File: source/module_esolver/esolver_dp.cpp [hunk 1]
```diff
@@ -75,16 +97,31 @@ namespace ModuleESolver
         cell[8] = ucell.latvec.e33 * ucell.lat0_angstrom;
 
         int iat = 0;
+
+        // added by Haocheng 2024/6/2 //
         for (int it = 0; it < ucell.ntype; ++it)
-        {
+        {   
+            #pragma omp parallel for schedule(static)
```
Comment (dyzheng) [inline source/module_esolver/esolver_dp.cpp]: Hello, I suggest to change loop of `it-ia` to the loop of `iat` , you can follow this code : https://github.com/deepmodeling/abacus-develop/blob/develop/source/module_hamilt_lcao/hamilt_lcaodft/operator_lcao/ekinetic_new.cpp#L108

**Review by dyzheng — CHANGES_REQUESTED** [review]

**Review by haozhihan — APPROVED** [review]
openmp part, LGTM

**Comment by HaCTang** [comment]
> openmp part, LGTM

To be frank, there are still some bugs because the wrong use of loop. Please see my latest version, which can shorten the time from 5s to 3s when using 16 cores.

File: source/module_esolver/esolver_dp.cpp (hunk 1 above)
Comment (HaCTang) [reply source/module_esolver/esolver_dp.cpp]: Please review my latest version.

**Comment by haozhihan** [comment]
> > openmp part, LGTM
> 
> 
> 
> To be frank, there are still some bugs because the wrong use of loop. Please see my latest version, which can shorten the time from 5s to 3s when using 16 cores.

Great！

**Comment by mohanchen** [comment]
It's a nice try! However, since no more updates, I will close this PR.

**Pull request closed without merging**